executor/extension/validator/eth_state_test_validator.go

```diff
@@ -76,23 +73,33 @@ func (e *ethStateTestValidator) PostTransaction(state executor.State[txcontext.T
 		return nil
 	}
 
-	if !e.cfg.ContinueOnFailure {
-		return err
-	}
-
-	ctx.ErrorInput <- err
-	e.finalErr = errors.Join(e.finalErr, err)
-	e.numberOfErrors++
+	return e.checkFatality(err, ctx.ErrorInput)
+}
 
-	// endless run
-	if e.cfg.MaxNumErrors == 0 {
+// PostBlock validates state root hash.
+// This needs to be done here instead of PostTransaction because EndBlock is being called in PostTransaction in
```

Comment:
We could move `BeginBlock` to `PreBlock` but we cannot move `EndBlock` to `PostBlock` because of the order. `PostTx` gets called before any `PostBlock` calls. If we moved `EndBlock` to `PostBlock` it would trigger after 
 `statedb.GetHash` which would cause the tool to fail validation.